esm-named-exports.test.js

```diff
@@ -7,8 +7,6 @@ const expect = chai.expect;
  * Context: https://github.com/sequelize/sequelize/pull/13689
  */
 
-const exportPaths = ['@sequelize/core', '@sequelize/core/decorators-legacy'];
```

Comment:
Can we move this file to `test/unit/esm-named-exports.test.js`? I don't really like this loose file in the root folder (same comment for `register-esbuild.js` which I prefer to have in `test/register-esbuild.js`)

Reply:
I moved it to `test/esm-named-exports.test.js`, no `/unit` since the test folder only contains two files